db_functions.py

```diff
@@ -232,7 +233,9 @@ def punycode_need_encode(orig):
     - contains a char that should be escaped or
     - ends with a dot or a space.
     """
-    if orig != escape_string(orig):
+    if not all(
+        (0x20 <= ord(c) < 0x80) and
```

Comment:
```suggestion
            (0x20 <= ord(c) < 0x80) and
```
E.g. please add one level of indentation here and on the line below, so it is visually distinct from the body of the condition.